src/lib/Libifl/pbsD_msgjob.c

```diff
@@ -229,27 +229,32 @@ char *extend;
 		char nd_ct_selstr[20];
 		char *endptr = NULL, *extend_dup = strdup(extend), *num_only;
 
-		num_only = strtok(extend_dup, " \'\"\n\t\v");
-		strtol(num_only, &endptr, 10);
-
-		if (num_only && !*endptr) {
-			snprintf(nd_ct_selstr, sizeof(nd_ct_selstr), "select=%s", num_only);
-			extend = nd_ct_selstr;
-		} else if ((i = set_resources(&attrib, extend, 1, &erp))) {
-			if (i > 1) {
-				snprintf(ebuff, sizeof(ebuff), "%s: %s\n", emsg_illegal_k_value, pbs_parse_err_msg(i));
-				emsg = strdup(ebuff);
-			} else
-				emsg = strdup("illegal -k value\n");
-			pbs_errno = PBSE_INVALSELECTRESC;
+		if (extend_dup) {
+			num_only = strtok(extend_dup, " \'\"\n\t\v");
+			strtol(num_only, &endptr, 10);
```

Comment:
This has gotten way more complicated than I intended. My initial request was that you check the return code of strtol(). Please allow me to clarify...

Please set errno to zero before the call and check it after. Also check whether the return value is LONG_MIN or LONG_MAX, per the man page. Finally, check whether *endptr points to '\0'. If everything looks good, then the parameter is a valid number. If not, then try the set_resources() call.

Checking for whitespace or quotes addresses a case where someone supplies -k '\'123  ' but does not address the example I provided. Confirming that strtol() ran properly is all you really need to do. No need to strdup() the argument.

Otherwise, things are looking good to me.

Reply:
Hope the changes in https://github.com/PBSPro/pbspro/pull/1452/commits/5c04242d3f819f0c4004c997443a79b4e0a5286d suffice. Now ditched the endptr dependency. Not checking the return value for LONG_MIN or LONG_MAX as the errno will be set anyways for overlfows and underflows.